docs/setup/metal3-setup.md

```diff
@@ -0,0 +1,225 @@
+# Intro 
+
+## Metal<sup>3</sup>
+
+Metal<sup>3</sup> (Metal-Kubed) is an open source tool that provides components for bare metal host management utilizing the Kubernetes native API. (see https://metal3.io/)
+
+### Pre-requisite Dependencies
+<details>
+  <summary>Click Here for list of Ubuntu Dependencies (22.04 LTS)</summary>
+  <br>
+
+  Make sure your packages are up to date
+  ```
+  sudo apt-get update -y
+  sudo apt-get dist-upgrade -y
+  ```
+  To avoid kernal upgrade messages, pass the `DEBIAN_FRONTEND=noninteractive` flag like so:
+  ```
+  sudo DEBIAN_FRONTEND=noninteractive apt-get update -y
+  sudo DEBIAN_FRONTEND=noninteractive apt-get dist-upgrade -y
+  ```
+  Otherwise, just press enter when it asks about the kernal updates.
+  
+  Dependencies: <br>
+  Make sure to have python3-pip installed first:
+  ```
+  sudo apt install python3-pip -y
+  ```
+  ```
+  python3 -m pip install ansible
+  sudo apt install libvirt-clients -y
+  sudo apt install qemu-kvm -y
+  sudo apt install libvirt-daemon-system -y
+  sudo apt install pkg-config -y
+  sudo apt install libvirt-dev -y
+  sudo apt install mkisofs -y
+  sudo apt install qemu -y
+  sudo apt install virtinst -y
+  sudo apt install qemu-efi -y
+  sudo apt install sshpass -y
+  pip3 install libvirt-python
+  ```
+  </details>
+
+
+## Deploying the SUSE Edge Metal<sup>3</sup> Demo
+1. Begin by cloning the SUSE Edge M3 Demo repo 
+```
+git clone https://github.com/suse-edge/metal3-demo.git
+```
+2. Copy the extra_vars.yml.example to extra_vars.yml
+```
+cp extra_vars.yml.example extra_vars.yml
+```
+
+3. Follow the instructions within extra_vars.yml to configure it according to your needs. If you would like a pre-configured file, skip to step 4, otherwise, skip to step 6. 
+- Important note: There are hardcoded memory and cpu configurations within `playbooks/setup_metal3_core.yaml` and `roles/vm/defaults/main.yaml` that are fairly intensive. You can decrease these resources to fit your environment, but do note that if the resources are too little, you may run into unexpected issues.
+- The lowest resource configuration that is confirmed to work is: 6 VCPUs and 16000 vm_memory in `playbooks/setup_metal3_core.yaml` and 8000 vm_memory in `roles/vm/defaults/main.yaml`.
+
+4. Show Pre-configured extra_vars.yml.
+<details>
+  <summary>Click here for a pre-configured extra_vars.yml file</summary>
+
+```yaml
+##
+# Whether to deploy sylva-core
+#
+deploy_sylva_core: false
+sylva_core_repo_url: https://gitlab.com/codefol/sylva-core.git
```

Comment:
This points to a personal fork from @guangyee - is that expected/desired?  

I'm wondering if we should instead be defaulting to the upstream repo, but I know  @dprodanov4 and @ipetrov117 have work in-progress to enable that in combination with the downstream charts.

It does look like the [default points to upstream](https://github.com/suse-edge/metal3-demo/blob/main/roles/baremetal/defaults/main.yml#L6), so perhaps just remove this line?

Reply:
Yeah this is expected
I'll have to look for the details later as this was a few months ago now but from what I can recall, Sylva upstream wouldn't approve a specific change that was necessary so Guang's fork is identical to upstream with one small change to allow everything to work.

Reply:
Ok thanks - from our conversation it seems it's due to https://gitlab.com/sylva-projects/sylva-core/-/merge_requests/257 - perhaps we can add a comment here to explain that (or do so as a follow-up) as it's not currently very clear if/why the fork is needed

Reply:
Just added a comment that shows when the sample configuration is expanded, let me know if this works with you